src/twister2/log_files/log_file.py

```diff
@@ -33,6 +34,24 @@ def _normalize_filename_path(filename: str) -> str:
         filename = os.path.normpath(os.path.abspath(filename))
         return filename
 
-    @abc.abstractmethod
-    def handle(self, data: str) -> None:
+    def handle(self, data: str | bytes) -> None:
         """Save information to logging file."""
+        if type(data) in [bytes, str]:
```

Comment:
We expect data to be string or bytes, this additional check will slow down the code several times. It is better to assume data is always string or bytes and remove this check which will be probably call hundreds of thousands times. 

I was testing something similar and these are my results
0.013081917000818066  # with type checking
0.0007284099992830306  # without type checking

Reply:
Done